### ISSUE TITLE
CheckBox - Providing a label on a FormField and on its contained Checkbox leads to an orphaned label on the FormField

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7325]
### Expected Behavior
Each FormField should have 1 label.

### Actual Behavior
Providing a label on a FormField and on its contained Checkbox leads to an orphaned label on the FormField. Once of the labels is not associated with anything.

Potentially RadioButton, RadioButtonGroup and CheckBoxGroup also have this issue

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by NOMADE55** [comment]
I could be wrong on what's being needed here, but I see an old part of the code where it checks if the prop component is `Checkbox`, and if it is, prevents `FormField` from adding a label.

![image](https://github.com/user-attachments/assets/7d1d3a13-1bd2-4bed-ba8c-0c0a1d471913)
